test/common/src/main/kotlin/org/codice/compliance/verification/profile/ProfilesVerifier.kt

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.verification.profile
+
+import org.codice.compliance.SAMLComplianceException
+import org.codice.compliance.SAMLSpecRefMessage.SAMLProfiles_3_1_a
+import org.codice.compliance.SAMLSpecRefMessage.SAMLProfiles_3_1_b
+import org.codice.compliance.SAMLSpecRefMessage.SAMLProfiles_3_1_c
+import org.codice.compliance.allChildren
+import org.codice.compliance.children
+import org.codice.compliance.utils.TestCommon.Companion.HOLDER_OF_KEY_URI
+import org.codice.compliance.utils.TestCommon.Companion.SAML_NAMESPACE
+import org.codice.compliance.utils.TestCommon.Companion.XSI
+import org.w3c.dom.Node
+
+class ProfilesVerifier(private val node: Node) {
+    /**
+     * Verify response against the Profiles Spec document
+     */
+    fun verify() {
+        verifyHolderOfKey()
+    }
+
+    private fun verifyHolderOfKey() {
+        val subjectConfirmationDataList = node.allChildren("SubjectConfirmation")
+                .filter { it.attributes.getNamedItem("Method").textContent == HOLDER_OF_KEY_URI }
+                .flatMap { it.children("SubjectConfirmationData") }
+
+        subjectConfirmationDataList.forEach {
+            val type = it.attributes.getNamedItemNS(XSI, "type")
+            if (type != null && !type.textContent.contains("KeyInfoConfirmationDataType"))
+                throw SAMLComplianceException.createWithPropertyNotEqualMessage(SAMLProfiles_3_1_b,
+                        "type", type.textContent, "KeyInfoConfirmationDataType")
+
+            if (type.firstChild.namespaceURI != SAML_NAMESPACE)
```

Comment:
❗️  This seems incorrect to me. What is the `firstChild` of the type attribute? Because we need to basically check what's before the colon in the string literal value. In this example, we need to check what namespace the `saml` prefix is pointing to:
`<SubjectConfirmationData xsi:type="saml:KeyInfoConfirmationDataType">`

Reply:
I had to debug and check that. The thing is if you get tj=he `value` it returns a string and you have no way of finding the namespace URL. `firstchild` actually returns the value with all the namespace information so we can verify it.

We can' just check what's before the colon because `saml` is a reference to `urn:oasis:names:tc:SAML:2.0:assertion`. DDF can use something else as long as it references the correct URI.

Reply:
Yeah I agree with you but there was nothing else I could find.

Reply:
Oh yeah. I have no issue with it now, as it is. I'm just commenting on how weird their API is 😄

Reply:
`tru` `fact`

Reply:
I **do** have issue with that ☝️ comment.